src/controllers/userProfileController.js

```diff
@@ -228,6 +230,11 @@ const userProfileController = function (UserProfile) {
         record.timeZone = req.body.timeZone;
         record.hoursByCategory = req.body.hoursByCategory;
         record.createdDate = moment(req.body.createdDate).toDate();
+        if (yearMonthDayDateValidator(req.body.endDate)) {
+          record.endDate = moment(req.body.endDate).toDate();
+        } else {
+          record.set('endDate', undefined, { strict: false });
```

Comment:
As for now, remove the endDate field if the data input is not valid. Currently the UI does not have the date validator and allows any invalid date input to submit but the UI will later get an error dialog.